hera_mc/autocorrelations.py

```diff
@@ -31,46 +33,97 @@ class _MeasurementTypes(object):
+    # recommended portable way of getting double-precision float.
+    value = Column(Float(precision="53"), nullable=False)
+
+    @classmethod
+    def create(cls, time, antenna_number, antenna_feed_pol, measurement_type, value):
+        """
+        Create a new Autocorrelation table object.
+
+        Parameters
+        ----------
+        time : astropy time object
+            Astropy time object based on timestamp of autocorrelation.
+        antenna_number : int
+            Antenna Number
+        antenna_feed_pol : str
+            Feed polarization, either 'e' or 'n'.
+        measurement_type : Int
+            The type of measurement as defined in the MeasurementTypes class.
+            Currently only supports 'median'.
+        value : float
+            The median autocorrelation value as a float.
+
+        """
+        if not isinstance(time, Time):
+            raise ValueError("time must be an astropy Time object.")
+        auto_time = floor(time.gps)
+
+        if antenna_feed_pol not in ['e', 'n']:
+            raise ValueError("antenna_feed_pol must be 'e' or 'n'.")
+
+        if isinstance(measurement_type, six.string_types):
+            try:
+                measurement_type = MeasurementTypes.names.index(measurement_type.lower())
+            except ValueError:
+                raise ValueError(
+                    "Autocorrelation type {0} not supported. "
+                    "Only the following types are supported: {1}"
+                    .format(measurement_type, MeasurementTypes.names)
+                )
 
-    measurement_type = NotNull(SmallInteger)
-    "The type of measurement; see MeasurementTypes enumeration."
+        if measurement_type not in list(six.moves.range(len(MeasurementTypes.names))):
+            raise ValueError(
+                "Input measurement type is not in range of accepted values. "
+                "Input {0}, Allowed range 0-{1}"
+                .format(measurement_type, len(MeasurementTypes.names) - 1)
+            )
 
-    # recommended portable way of getting double-precision float.
-    value = NotNull(Float(precision='53'))
-    "The autocorrelation value."
+        return cls(time=auto_time, antenna_number=antenna_number,
+                   antenna_feed_pol=antenna_feed_pol, measurement_type=0, value=value)
 
     @property
     def measurement_type_name(self):
         return MeasurementTypes.names[self.measurement_type]
 
     def __repr__(self):
-        return('<Autocorrelations id={self.id} time={self.time} antnum={self.antnum} '
-               'polarization={self.polarization} measurement_type={self.measurement_type_name} '
-               'value={self.value}>').format(self=self)
+        return("<HeraAuto time={self.time} antenna_number={self.antenna_number} "
+               "polarization={self.antenna_feed_pol} measurement_type={self.measurement_type_name} "
+               "value={self.value}>").format(self=self)
```

Comment:
There is a standard representation defined for MCDeclarativeBase objects, so only do this if that one doesn't do the right thing.

Reply:
Oh I didn't know that! This was still carry over from the original table.